wandb/sdk/wandb_settings.py

```diff
@@ -713,6 +713,13 @@ def _apply_source_login(
             _logger.info("setting login settings: {}".format(login_settings))
         self._update(login_settings, _source=self.Source.LOGIN)
 
+    def _apply_setup(
+        self, setup_settings: Dict[str, Any], _logger: Optional[_EarlyLogger] = None
+    ) -> None:
+        if _logger:
+            _logger.info("setting setup settings: {}".format(setup_settings))
+        self._update(setup_settings, _source=self.Source.SETUP)
```

Comment:
using Source.SETUP as it's currently unused & talked to jeff about it, looking into this long term probably not the best place for this but other Source options can cause unexpected interactions w/other things

Reply:
I think this can be refactored in the settings project that I'll get back to soon. I'm working on adding better test infra for it as well.

Reply:
Looks good.
Not that i seee the code, just for clarity sake, can you make a newsource between LOGIN and INIT, 
maybe Source.INTEGRATION or Source.SAGE.

and call the method _apply_source_integration or (something similar)

Reply:
we sent this branch to the user already so would like to keep it stable but yeah we can add one in after next release?

Reply:
Are you waiting for feedback from the user? it would be good to resync with master and get this merged (which would change this branch)

If you are worried about changing the branch, you can merge this into staging branch do any master merging and other trivial changes there, then merge it into master.

Reply:
im fine with delaying the SOURCE change, it is trivial and mainly for readability